src/coreclr/tools/Common/JitInterface/CorInfoImpl.cs

```diff
@@ -2354,6 +2352,133 @@ private uint getClassNumInstanceFields(CORINFO_CLASS_STRUCT_* cls)
             throw new InvalidOperationException();
         }
 
+        private GetTypeLayoutResult GetTypeLayoutHelper(MetadataType type, uint parentIndex, uint baseOffs, FieldDesc field, CORINFO_TYPE_LAYOUT_NODE* treeNodes, nuint maxTreeNodes, nuint* numTreeNodes)
+        {
+            if (*numTreeNodes >= maxTreeNodes)
+            {
+                return GetTypeLayoutResult.Partial;
+            }
+
+            uint structNodeIndex = (uint)(*numTreeNodes)++;
+            CORINFO_TYPE_LAYOUT_NODE* parNode = &treeNodes[structNodeIndex];
+            parNode->typeHnd = ObjectToHandle(type);
+            parNode->fieldHnd = field == null ? null : ObjectToHandle(field);
+            parNode->parent = parentIndex;
```

Comment:
Capturing the field and type handles here is problematic for crossgen2 and ready to run. Can you describe what these are actually used for?

Reply:
The type handle is used for promoting SIMD types to JIT primitives -- I can change the name to `simdTypeHnd` and only assign it in that path.
The field handle is only used for diagnostic purposes (attaching field names to the promoted locals in debug output). I wasn't sure how to avoid this one in a lazy way.

Reply:
diagnostic purposes only is completely ok, although please name the field so that folks know not to use it for anything complex.

For the promotion of SIMD types to JIT primitives, do we ever pass the typeHnd acquired here back into the VM? Could we just put some sort of enum or something instead of a full type handle?

Reply:
Talked with @davidwrighton offline and ended up changing this to only be for SIMD types plus adding extra guards/documentation on what is permitted with the type handle.